src/Tools/TestDiscoveryWorker/Program.cs

```diff
@@ -0,0 +1,118 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.IO.Pipes;
+using System.Linq;
+using System.Reflection;
+using System.Text.Json;
+using System.Threading.Channels;
+
+using Xunit;
+using Xunit.Abstractions;
+
+if (args.Length <= 0)
+{
+    return;
+}
+
+using var pipeClient = new AnonymousPipeClientStream(PipeDirection.In, args[0]);
+using var sr = new StreamReader(pipeClient);
+// Display the read text to the console
+string? output;
+
+// Wait for 'sync message' from the server.
+do
+{
+    output = sr.ReadLine();
+}
+while (!(output?.StartsWith("ASSEMBLY") == true));
```

Comment:
Use an explicit string comparer whenever using `StartsWith`

Reply:
In the case `"ASSEMBLY"` is not returned this will just infinitely loop because `null` fails the condition.